src/app/pages/data-protection/snapshot-task/snapshot-task-card/snapshot-task-card.component.ts

```diff
@@ -0,0 +1,145 @@
+      propertyName: 'next_run',
+    }),
+    toggleColumn({
+      title: this.translate.instant('Enabled'),
+      propertyName: 'enabled',
+      cssClass: 'justify-end',
+      onRowToggle: (row: PeriodicSnapshotTaskUi) => this.onChangeEnabledState(row),
+    }),
+    stateButtonColumn({
+      title: this.translate.instant('State'),
+      getValue: (row) => row.state.state,
+      cssClass: 'state-button',
+    }),
+    textColumn({
+      propertyName: 'id',
+    }),
+  ]);
+
+  constructor(
+    private slideInService: IxSlideInService,
+    private translate: TranslateService,
+    private errorHandler: ErrorHandlerService,
+    private ws: WebSocketService,
+    private dialogService: DialogService,
+    private cdr: ChangeDetectorRef,
+    private taskService: TaskService,
+  ) {}
+
+  ngOnInit(): void {
+    this.getSnapshotTasks();
+  }
+
+  getSnapshotTasks(): void {
+    this.isLoading = true;
+    this.ws.call('pool.snapshottask.query').pipe(
+      untilDestroyed(this),
+    ).subscribe((snapshotTasks: PeriodicSnapshotTaskUi[]) => {
+      const transformedSnapshotTasks = this.transformSnapshotTasks(snapshotTasks);
+      this.snapshotTasks = transformedSnapshotTasks;
+      this.dataProvider.setRows(transformedSnapshotTasks);
+      this.isLoading = false;
+      this.cdr.markForCheck();
+    });
+  }
+
+  doDelete(snapshotTask: PeriodicSnapshotTaskUi): void {
+    this.dialogService.confirm({
+      title: this.translate.instant('Confirmation'),
+      message: this.translate.instant('Delete Periodic Snapshot Task <b>"{value}"</b>?', {
+        value: `${snapshotTask.dataset} - ${snapshotTask.naming_schema} - ${snapshotTask.keepfor}`,
+      }),
+    }).pipe(
+      filter(Boolean),
+      switchMap(() => this.ws.call('pool.snapshottask.delete', [snapshotTask.id])),
+      untilDestroyed(this),
+    ).subscribe({
+      next: () => {
+        this.getSnapshotTasks();
+      },
+      error: (err) => {
+        this.dialogService.error(this.errorHandler.parseError(err));
+      },
+    });
+  }
+
+  openForm(row?: PeriodicSnapshotTaskUi): void {
+    const slideInRef = this.slideInService.open(SnapshotTaskFormComponent, { data: row, wide: true });
+
+    slideInRef.slideInClosed$.pipe(untilDestroyed(this)).subscribe(() => {
+      this.getSnapshotTasks();
+    });
+  }
+
+  private onChangeEnabledState(snapshotTask: PeriodicSnapshotTaskUi): void {
+    this.ws
+      .call('pool.snapshottask.update', [snapshotTask.id, { enabled: !snapshotTask.enabled } as PeriodicSnapshotTaskUi])
+      .pipe(untilDestroyed(this))
+      .subscribe({
+        next: () => {
+          this.getSnapshotTasks();
+        },
+        error: (err: WebsocketError) => {
+          this.getSnapshotTasks();
+          this.dialogService.error(this.errorHandler.parseWsError(err));
+        },
+      });
+  }
+
+  private transformSnapshotTasks(snapshotTasks: PeriodicSnapshotTaskUi[]): PeriodicSnapshotTaskUi[] {
```

Comment:
Since this PR https://github.com/truenas/webui/pull/8841/files, there is an option to use `getValue` in column definitions. I'd rather we use that than custom interfaces like `PeriodicSnapshotTaskUi`.